rust/types/src/task.rs

```diff
@@ -26,6 +26,12 @@ impl From<AttachedFunctionUuid> for JobId {
     }
 }
 
+impl From<NonceUuid> for JobId {
```

Comment:
I don't think this is necessary. We should be storing AttachedFunctionIds in the in_progress_jobs map because we want the scorecarding to be scoped to the attached function.